src/pip/_internal/commands/configuration.py

```diff
@@ -127,27 +138,39 @@ def run(self, options, args):
         return SUCCESS
 
     def _determine_file(self, options, need_value):
-        file_options = {
-            kinds.USER: options.user_file,
-            kinds.GLOBAL: options.global_file,
-            kinds.VENV: options.venv_file
-        }
-
-        if sum(file_options.values()) == 0:
+        # Convert legacy venv_file option to site_file or error
+        if options.venv_file and not options.site_file:
+            if running_under_virtualenv():
+                options.site_file = True
+                logger.warning(
+                    "The --venv option is deprecated. Use --site instead."
+                )
+            else:
+                raise PipError(
+                    "Legacy --venv option requires a virtual environment. "
+                    "Use --site instead."
+                )
+
+        file_options = [key for key in [
+            kinds.USER if options.user_file else None,
+            kinds.GLOBAL if options.global_file else None,
+            kinds.SITE if options.site_file else None,
+        ] if key]
```

Comment:
FWIW, this was confusing for me to understand at first. I think part of it (aside from being a nested comprehension) is that it's doing a comparison on each item twice (first to `None`, and then you check each `None`) when conceptually you should only need one. An alternative would just be to do:

```python
file_options = [key for key, value in (
    (kinds.USER, options.user_file),
    (kinds.GLOBAL, options.global_file),
    (kinds.SITE, options.site_file),
) if value]
```